### PR TITLE
Add an additional test and code cleanup

### PR DESCRIPTION
**Pull request by Joy603**
Changes include one additional selenium tests and some code cleanup because we are refactoring code from selenium_common modules.

### CODE REVIEW

File: canvas_account_admin_tools/requirements/local.txt [hunk 1]
```diff
@@ -20,3 +20,4 @@ pyvirtualdisplay
 # icommons_common and into ccsw, work around the circular dependency by calling
 # out the ccsw dependency here.
 git+ssh://git@github.com/Harvard-University-iCommons/django-canvas-course-site-wizard.git@v1.3.8#egg=django-canvas-course-site-wizard==1.3.8
+git+ssh://git@github.com/Harvard-University-iCommons/selenium_common.git@develop#egg=selenium-common
```
Comment (rascalking) [inline canvas_account_admin_tools/requirements/local.txt]: We should figure out a versioning scheme for selenium-common, so we don't hit version mismatch issues.

File: selenium_tests/course_info/course_info_people_list_test.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        search_page.is_loaded()
+
+        # submit a search term, a course instance id in this case
+        search_page.submit_search("347547")
+
+        # click on course link to view list of people in course
+        search_page.select_course_id()
+
+        # assert that this person is in this course
+        self.assertTrue(self.driver.find_element_by_xpath("//td[contains(text(), '20881755')]"))
```
Comment (rascalking) [inline selenium_tests/course_info/course_info_people_list_test.py]: This feels like it should be a method on the page object (the find td bit, I mean).

File: selenium_tests/course_info/course_info_people_list_test.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+        # initialize
+        parent_page = AccountAdminDashboardPage(self.driver)
+        # navigate to course info page
+        search_page = parent_page.select_course_info_link()
+
+        # check if page is loaded(which will also set the focus on the tool), before selecting search terms
+        search_page.is_loaded()
+
+        # submit a search term, a course instance id in this case
+        search_page.submit_search("347547")
```
Comment (rascalking) [inline selenium_tests/course_info/course_info_people_list_test.py]: Stick the instance ID in a constant somewhere?

File: selenium_tests/regression_suite.py [hunk 4]
```diff
@@ -17,35 +17,43 @@
...
+    def main(self):
+
+        date_timestamp = time.strftime('%Y%m%d_%H_%M_%S')
+
+        # This relative path should point to BASE_DIR/selenium_tests/reports
+        report_file_path = path.relpath('./reports')
+        if not path.exists(report_file_path):
+            makedirs(report_file_path)
+        report_file_name = "course_info_test_report_{}.html".format(date_timestamp)
+        report_file_buffer = file(path.join(report_file_path, report_file_name), 'wb')
```
Comment (rascalking) [inline selenium_tests/regression_suite.py]: Naming nitpick...it's not a buffer, it's a file object.

File: selenium_tests/regression_suite.py [hunk 5]
```diff
@@ -17,35 +17,43 @@
...
+        runner = HTMLTestRunner.HTMLTestRunner(
+            stream=report_file_buffer,
+            title='Course Info test suite report',
+            description='Result of tests in {}'.format(__file__)
+        )
+
+        # course search - test the flow of the app from login to course search
+        dashboard_page_tests = unittest.TestLoader().loadTestsFromTestCase(AccountAdminIsDasboardLoadedTest)
```
Comment (rascalking) [inline selenium_tests/regression_suite.py]: I know performance doesn't *really* matter here, but there's no need to create and throw away a loader object for each time we load tests from a testcase.

File: selenium_tests/regression_suite.py [hunk 6]
```diff
@@ -17,35 +17,43 @@
...
+        course_info_initialize_testing = unittest.TestLoader().loadTestsFromTestCase(CourseInfoIsSearchPageLoadedTest)
+        course_info_search_testing = unittest.TestLoader().loadTestsFromTestCase(CourseInfoSearchTest)
+        course_info_people_testing = unittest.TestLoader().loadTestsFromTestCase(CourseInfoPeopleListTest)
+
+        # create a test suite combining the tests above
+        smoke_tests = unittest.TestSuite([dashboard_page_tests, course_info_initialize_testing, course_info_search_testing, course_info_people_testing])
```
Comment (rascalking) [inline selenium_tests/regression_suite.py]: Is there a reason we're manually importing and loading the test cases, then sticking them in a test suite by hand?  Can we not just let it discover the testcases for us?

https://github.com/Harvard-ATG/canvas_course_info/blob/develop/selenium_tests/regression_suite.py#L35

**Comment by rascalking** [comment]
:+1:

File: selenium_tests/regression_suite.py (hunk 6 above)
Comment (elliottyates) [inline selenium_tests/regression_suite.py]: If test cases are always written to be idempotent, we can hopefully just let the unittest framework find and run them.

Comment (Joy603) [inline selenium_tests/regression_suite.py]: Love it.  Done.

File: canvas_account_admin_tools/requirements/local.txt (hunk 1 above)
Comment (Joy603) [inline canvas_account_admin_tools/requirements/local.txt]: Done

File: selenium_tests/course_info/course_info_people_list_test.py (hunk 2 above)
Comment (Joy603) [inline selenium_tests/course_info/course_info_people_list_test.py]: Done

File: selenium_tests/course_info/course_info_people_list_test.py (hunk 3 above)
Comment (Joy603) [inline selenium_tests/course_info/course_info_people_list_test.py]: Done

File: selenium_tests/regression_suite.py (hunk 4 above)
Comment (Joy603) [inline selenium_tests/regression_suite.py]: Re-named

File: selenium_tests/regression_suite.py (hunk 5 above)
Comment (Joy603) [inline selenium_tests/regression_suite.py]: Fixed

**Pull request merged**